src/common/tusb_compiler.h

```diff
@@ -67,20 +70,80 @@
 
   // Endian conversion use well-known host to network (big endian) naming
   #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-    #define tu_htonl(u32)  __builtin_bswap32(u32)
-    #define tu_ntohl(u32)  __builtin_bswap32(u32)
-
-    #define tu_htons(u16)  __builtin_bswap16(u16)
-    #define tu_ntohs(u16)  __builtin_bswap16(u16)
+    #define TU_BYTE_ORDER TU_LITTLE_ENDIAN
   #else
-    #define tu_htonl(u32)  (u32)
-    #define tu_ntohl(u32)  (u32)
+    #define TU_BYTE_ORDER TU_BIG_ENDIAN
+  #endif
+
+  static inline uint16_t tu_bswap16(uint16_t u16)
+  {
+    return __builtin_bswap16(u16);
+  }
+
+  static inline uint16_t tu_bswap32(uint16_t u32)
+  {
+    return __builtin_bswap32(u32);
+  }
+
+  #define TU_BSWAP16
+
+#elif defined(__TI_COMPILER_VERSION__)

```

Comment:
I'm guessing this isn't the GCC compiler, but FWIW I develop using `msp430-elf-gcc`, and it _has_ `__BYTE_ORDER__` defined:

```
William@William-THINK MINGW64 ~/src/tinyusb
$ msp430-elf-gcc -dM -E - < /dev/null | grep __BYTE_ORDER__
#define __BYTE_ORDER__ __ORDER_LITTLE_ENDIAN__

William@William-THINK MINGW64 ~/src/tinyusb
$ msp430-elf-gcc -v
Using built-in specs.
COLLECT_GCC=C:\msys64\opt\msp430-elf-toolchain\bin\msp430-elf-gcc.exe
COLLECT_LTO_WRAPPER=c:/msys64/opt/msp430-elf-toolchain/bin/../libexec/gcc/msp430-elf/5.3.0/lto-wrapper.exe
Target: msp430-elf
Configured with: /srv/net/msp430/jenkins/workspace/msp430-pkg-gnu/gcc.subtree/configure --build=x86_64-redhat-linux --host=i686-w64-mingw32 --target=msp430-elf --enable-languages=c,c++ --disable-nls --enable-target-optspace --enable-newlib-nano-formatted-io --with-pkgversion='SOMNIUM Technologies Limited - msp430-gcc 5.3.0.219'
Thread model: single
gcc version 5.3.0 (SOMNIUM Technologies Limited - msp430-gcc 5.3.0.219)
```

Of course, being GCC, the `__GNUC__` logic will trigger :).

Reply:
As intended. I think.... I wasn't sure if you were going to use the TI or gcc compiler. I mostly use CCS for programming.

Reply:
@pigrew Does the CCS's TI compiler define `__GNUC__` as well, everything looks pretty much the same as GNUC. If yes, perhaps we could merge them together ?

Reply:
No, it does not define __GNUC__.

The logic of endianness selection is different, but they do support most of the GCC macros

[TI ARM compiler docs](http://www.ti.com/lit/ug/spnu151r/spnu151r.pdf)
[MSP430 compiler docs](http://www.ti.com/lit/ug/slau132r/slau132r.pdf)